### PR TITLE
Authentication Redesign

### PR DESCRIPTION
**Pull request by cmckee-dev**
**User Sign In:**  

user-sign-in

---
**User Forgot Password:**  

user-forgot-password

---
**User Resend Confirmation:**

user-confirmation-resend

---
**User Resend Unlock:**  

user-unlock-account

---
**Update User Password:**  

update-password

---
**Invite User:**  

invite-new-user

---

There is a bug with the footer / containers where if the page content doesn't fill the page the footer floats up exposing white space. I temporarily set footer position to absolute while working on auth redesign. We will need to address this before redesign is finished.  

Screenshot_2019-11-04 Sessions

### CODE REVIEW

File: app/views_redesigned/devise/confirmations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<%= render "devise/layout" do %>
+  <%= simple_form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { class: "auth-form js-validate mx-auto", method: :post }) do |f| %>
```
Comment (andrewmcodes) [inline app/views_redesigned/devise/confirmations/new.html.erb]: not sure we need this `js-validate` class. May want to check on that but I think that's from the old template.

File: app/views_redesigned/devise/confirmations/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    <div class="text-center mb-2">
+      <h4 class="font-weight-normal">Resend <span class="text-success">Confirmation</span>?</h4>
+      <p>Enter your email and let's get you back on track.</p>
+    </div>
+    <div class="js-form-message form-group">
```
Comment (andrewmcodes) [inline app/views_redesigned/devise/confirmations/new.html.erb]: Same with `js-form-message`

File: app/views_redesigned/devise/confirmations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                label: false,
+                autofocus: true,
+                value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email),
+                input_html: { autocomplete: "email" },
+                placeholder: "Email address",
+                data: {
+                  rule_email: true,
+                  msg: "Please enter a valid email address.",
+                  error_class: "u-has-error",
+                  success_class: "u-has-success"
```
Comment (andrewmcodes) [inline app/views_redesigned/devise/confirmations/new.html.erb]: these `u-` classes no longer exist.

File: app/views_redesigned/devise/passwords/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+<%= render "devise/layout" do %>
+  <%= simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post, class: "auth-form js-validate mx-auto" }) do |f| %>
+    <div class="text-center mb-2">
+      <h4 class="font-weight-normal">Forgot your <span class="text-success">Password</span>?</h4>
+      <p>Enter your email and let's get you back on track.</p>
+    </div>
+
+    <div class="js-form-message form-group">
```
Comment (andrewmcodes) [inline app/views_redesigned/devise/passwords/new.html.erb]: more `js-` and `u-` clases in here.

File: app/views_redesigned/devise/sessions/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+              input_html: {
+                class: "form-control",
+                autocomplete: "email",
+                tabindex: 1,
+              },
+              data: {
+                rule_email: true,
+                msg: "Please enter a valid email address.",
+                errorClass: "u-has-error",
+                successCass: "u-has-success"
```
Comment (andrewmcodes) [inline app/views_redesigned/devise/sessions/new.html.erb]: more `js-` and `u-` classes.

File: app/views_redesigned/user_passwords/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<%= card do %>
+  <%= card_body do %>
+    <%= form_tag user_passwords_path, method: :put, class: "js-validate mx-auto" do %>
+      <div class="js-form-message js-focus-state mb-3">
```
Comment (andrewmcodes) [inline app/views_redesigned/user_passwords/_form.html.erb]: more `js-` and `u-` classes

**Comment by andrewmcodes** [comment]
also make sure you run `bin/standardize`

File: app/views_redesigned/user_passwords/_form.html.erb (hunk 6 above)
Comment (cmckee-dev) [reply app/views_redesigned/user_passwords/_form.html.erb]: :heavy_check_mark:

File: app/views_redesigned/devise/sessions/new.html.erb (hunk 5 above)
Comment (cmckee-dev) [reply app/views_redesigned/devise/sessions/new.html.erb]: :heavy_check_mark:

File: app/views_redesigned/devise/passwords/new.html.erb (hunk 4 above)
Comment (cmckee-dev) [reply app/views_redesigned/devise/passwords/new.html.erb]: :heavy_check_mark:

File: app/views_redesigned/devise/confirmations/new.html.erb (hunk 3 above)
Comment (cmckee-dev) [reply app/views_redesigned/devise/confirmations/new.html.erb]: :heavy_check_mark:

File: app/views_redesigned/devise/confirmations/new.html.erb (hunk 2 above)
Comment (cmckee-dev) [reply app/views_redesigned/devise/confirmations/new.html.erb]: :heavy_check_mark:

File: app/views_redesigned/devise/confirmations/new.html.erb (hunk 1 above)
Comment (cmckee-dev) [reply app/views_redesigned/devise/confirmations/new.html.erb]: :heavy_check_mark:

**Pull request merged**